crypto/stateproof/msgp_gen.go

```diff
@@ -1409,6 +1458,26 @@ func (z *StateProof) MsgIsZero() bool {
 	return ((*z).SigCommit.MsgIsZero()) && ((*z).SignedWeight == 0) && ((*z).SigProofs.MsgIsZero()) && ((*z).PartProofs.MsgIsZero()) && ((*z).MerkleSignatureSaltVersion == 0) && (len((*z).Reveals) == 0) && (len((*z).PositionsToReveal) == 0)
 }
 
+// MaxSize returns a maximum valid message size for this message type
+func StateProofMaxSize() (s int) {
+	s = 1 + 2 + crypto.GenericDigestMaxSize() + 2 + msgp.Uint64Size + 2
+	// Using totalallocbound for: z.SigProofs
+	s += SigPartProofMaxSize
+	s += 2
+	// Using totalallocbound for: z.PartProofs
+	s += SigPartProofMaxSize
```

Comment:
How is this generated? We have `s += SigPartProofMaxSize` twice.

Reply:
Both SigProofs and PartProofs have the same theoretical upper bound so are using the same allocbound constant in `StateProof` struct definition here: https://github.com/iansuvak/go-algorand/blob/3289d45fd12d78481d93f0b0000d0e461fbcc52e/crypto/stateproof/structs.go#L75-L76

Reply:
My bad. Thanks. You very elegantly have also generated the comments.